src/provider/element-templates/ReplaceBehavior.js

```diff
@@ -6,37 +10,47 @@ import { is } from 'bpmn-js/lib/util/ModelUtil';
  * if the element's type or its parent's is in
  * the list of elements the template applies to.
  */
-export default function ReplaceBehavior(elementTemplates, eventBus) {
-  eventBus.on('moddleCopy.canCopyProperty', function(context) {
-    const {
-      parent,
-      property,
-      propertyName
-    } = context;
-
-    if (propertyName !== 'modelerTemplate') {
-      return;
-    }
+export default function ReplaceBehavior(elementTemplates, eventBus, injector) {
+  injector.invoke(CommandInterceptor, this);
+
+  this.postExecuted('shape.replace', function(e) {
+    var context = e.context,
+        newShape = context.newShape,
+        newBo = newShape.businessObject;
+
+    const template = newBo.modelerTemplate;
+    const version = newBo.modelerTemplateVersion;
 
-    const elementTemplate = elementTemplates.get(property);
+    const elementTemplate = elementTemplates.get(template, version);
 
     if (!elementTemplate) {
-      return false;
+      removeELementTemplate(newBo);
+      return;
     }
 
     const { appliesTo } = elementTemplate;
 
     const allowed = appliesTo.reduce((allowed, type) => {
-      return allowed || is(parent, type);
+      return allowed || is(newBo, type);
     }, false);
 
     if (!allowed) {
-      return false;
+      removeELementTemplate(newBo);
```

Comment:
We should use the existing `updateProperties` handler here, as we don't always know if postExecuted works on a new element (does here) or not (needs to be undone) properly.

Reply:
I just found the `unlinkTemplate` utility that does exactly that :smile:

Reply:
To ensure we use the correct utility with the right prefix, I had to mirror the code for cloud templates (like we do with all element templates code)

Reply:
> like we do with all element templates code

Yes, but we recently tried to make this stuff more configurable (when possible). E.g. [the templates group](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/src/provider/element-templates/components/ElementTemplatesGroup.js#L60).

What we could consider is that the `elementTemplates` feature has a method `unlinkTemplate` / `removeTemplate`, so that the behavior can simply re-use this, e.g. what we have with [`applyTemplate`](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/src/provider/element-templates/ElementTemplates.js#L123).

Not a hard requirement, but could help us in different places.

Reply:
I'd like to keep this as a `fix`, and tackle the refactoring in a follow up

Reply:
💯 👍